sdks/python/src/opik/console_utils.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable
+import logging
+import tqdm
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _in_jupyter_environment() -> bool:
+    """
+    Check to see if code is running in a Jupyter environment,
+    including jupyter notebook, lab, or console.
+    """
+    try:
+        import IPython
+    except Exception:
+        return False
+
+    ipy = IPython.get_ipython()
+    if ipy is None or not hasattr(ipy, "kernel"):
+        return False
+    else:
+        return True
+
+
+def _in_ipython_environment() -> bool:
+    """
+    Check to see if code is running in an IPython environment.
+    """
+    try:
+        import IPython
+    except Exception:
+        return False
+
+    ipy = IPython.get_ipython()
+    if ipy is None:
+        return False
+    else:
+        return True
+
+
+def _in_colab_environment() -> bool:
+    """
+    Check to see if code is running in Google colab.
+    """
+    try:
+        import IPython
+    except Exception:
+        return False
+
+    ipy = IPython.get_ipython()
+    return "google.colab" in str(ipy)
+
+
+def get_tqdm() -> Callable:
```

Comment:
Let's use real tqdm types as output type hints here, otherwise we won't have code completions

Reply:
The return is one of two functions, so I don't think we can do that if I understand you correctly.

Reply:
Oh, I see. Fixed!